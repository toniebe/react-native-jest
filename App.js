import {Text, View, TouchableOpacity} from 'react-native';
import React, {Component} from 'react';
import Button from './src/components/ButtonCustom';

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      result: 0,
    };
  }

  sum = (a, b) => {
    this.setState({
      result: a + b,
    });
  };

  render() {
    return (
      <View>
        <Text testID='myTextResult'>{this.state.result}</Text>
        <Button />
        <TouchableOpacity testID='myButtonSum' onPress={() => this.sum(3,9)}>
          <Text>Sum</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export default App;
