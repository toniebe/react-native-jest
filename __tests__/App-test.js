/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer,{act} from 'react-test-renderer';

const tree = renderer.create(<App />)


it('renders correctly', () => {
  renderer.create(<App />);
});

test('renders App', () => {
  expect(tree).toMatchSnapshot();
});


test('button pressed', () => {
  // press button
  const button  = tree.root.findByProps({ testID: 'myButtonSum' }).props;
  act(() => button.onPress());
  
  const text  = tree.root.findByProps({ testID: 'myTextResult' }).props;
  expect(text.children).toEqual(12)
})

