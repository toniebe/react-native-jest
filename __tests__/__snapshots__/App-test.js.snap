// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<View>
  <Text
    testID="myTextResult"
  >
    0
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderWidth": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Button
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="myButtonSum"
  >
    <Text>
      Sum
    </Text>
  </View>
</View>
`;
